src/main/java/bsh/BSHArrayInitializer.java

```diff
@@ -109,9 +110,10 @@ public Object eval( Class<?> baseType, int dimensions, CallStack callstack,
             inferType = inferCommonType(null, this, callstack, interpreter);
 
         // force MapEntry to Map output
-        if ( MapEntry.class == inferType && Void.TYPE == baseType
-                || MapEntry.class == baseType )
-            baseType = Map.class;
+        if (dimensions < 2)
+            if ( MapEntry.class == inferType && Void.TYPE == baseType
+                 || MapEntry.class == baseType )
```

Comment:
Seems like a fair suggestion, I know the code didn't change but it came under review now.

Reply:
Reference to Lift's suggestion, now seems to be lost.

> [OperatorPrecedence](https://errorprone.info/bugpattern/OperatorPrecedence): Use grouping parenthesis to make the operator precedence explicit

Reply:
Ok, added